### DISCUSSION TITLE
Any chance grammars come with description of types?

### DISCUSSION THREAD

**Discussion by ag91** [tree-sitter/tree-sitter#1648] [Q&A] [answered]
Hi,

Tree sitter is amazing! I was wondering if grammar developers can provide an explanation of rules within tree-sitter.
For example, in an XML grammar I have `quoted_attribute_value` as a type of a node.
It would be cool if I could do something like: `tree-sitter explain quoted_attribute_value` and get a description and an example by the grammar author (or even a reference to another resource)

Is this possible already somehow? And if not, how do you find out what these types mean if you have no idea? Only by example?

Thanks!

**Comment by ObserverOfTime** [comment, marked as answer]
It's up to the author to document the nodes in `grammar.js`.
